flang/lib/Lower/ConvertVariable.cpp

```diff
@@ -1156,14 +1156,14 @@ void Fortran::lower::mapSymbolAttributes(
   const bool isDeclaredDummy = Fortran::semantics::IsDummy(sym);
   // An active dummy from the current entry point.
   const bool isDummy = isDeclaredDummy && symMap.lookupSymbol(sym).getAddr();
+  // An unused dummy from another entry point.
+  const bool isUnusedEntryDummy = isDeclaredDummy && !isDummy;
   const bool isResult = Fortran::semantics::IsFunctionResult(sym);
   const bool replace = isDummy || isResult;
   fir::factory::CharacterExprHelper charHelp{builder, loc};
 
   if (Fortran::semantics::IsProcedure(sym)) {
-    assert(isDeclaredDummy && "expected a dummy procedure argument");
-    if (!isDummy) {
-      // This is an unused dummy procedure argument from another entry point.
+    if (isUnusedEntryDummy) {
```

Comment:
Could you add a `if (`Fortran::semantics::IsPointer(sym)`) TODO(loc, "procedure pointer");`  just after `if (Fortran::semantics::IsProcedure(sym))`  ?
I did not notice in the previous review, but the early return for Procedure that was introduced prevent us from ever hitting the procedure pointer TODO for non dummy procedure pointers.

```
 procedure(), pointer :: foo
 call foo()
end
```
Hits : `failed to find indirect call symbol address` instead of a not yet implemented.

Reply:
done; thanks